scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -120,7 +120,11 @@ class AstNamerMacros(val c: Context) extends AstReflection with CommonNamerMacro
         val binaryCompatStats = binaryCompatVars.flatMap { case vr: ValDef =>
           List(
             q"def ${getterName(vr)} = this.${vr.name} ",
-            q"def ${setterName(vr)}(${vr.name} : ${vr.tpt}) = this.${vr.name} = ${vr.name} ",
+            q"""def ${setterName(vr)}(${vr.name} : ${vr.tpt}) = {
+                   val node = this
+                   $CommonTyperMacrosModule.storeField(this.${vr.name}, ${vr.name}, ${vr.name
```

Comment:
This actually copies the entire subtree for the field, which is normally done for other fields, but was not for binaryCompat ones.